transform_stats_to_metrics_test.go

```diff
@@ -1024,30 +1026,90 @@ func TestViewDataToMetrics_MissingVsEmptyLabelValues(t *testing.T) {
 	startTime := time.Date(2018, 11, 25, 15, 38, 18, 997, time.UTC)
 	endTime := startTime.Add(100 * time.Millisecond)
 
+	latencyView := &view.View{
+		Name:        "ocagent.io/latency",
+		Description: "speed of the various runners",
+		Aggregation: view.Sum(),
+		TagKeys:     []tag.Key{keyField, keyName, keyPlayerName},
+		Measure:     mSprinterLatencyMs,
+	}
+
+	view.Register(latencyView)
+	defer view.Unregister(latencyView)
+
+	stats.RecordWithTags(context.Background(), []tag.Mutator{
+		tag.Upsert(keyField, "main-field"),
+		tag.Upsert(keyName, "sprint-1"),
+		tag.Upsert(keyPlayerName, "player-1"),
+	}, mSprinterLatencyMs.M(2))
+	stats.RecordWithTags(context.Background(), []tag.Mutator{
+		tag.Upsert(keyField, "main-field"),
+		// Missing Sprint Name
+		tag.Upsert(keyPlayerName, "player-2"),
+	}, mSprinterLatencyMs.M(4))
+	stats.RecordWithTags(context.Background(), []tag.Mutator{
+		tag.Upsert(keyField, ""), // Empty field name
+		tag.Upsert(keyName, ""),  // Empty sprint name
+		// Duplicate tag values. Only the most recent tag value should be used.
+		tag.Upsert(keyPlayerName, "player-x"),
+		tag.Upsert(keyPlayerName, "player-3"),
+	}, mSprinterLatencyMs.M(6))
+	stats.RecordWithTags(context.Background(), []tag.Mutator{}, mSprinterLatencyMs.M(8))
```

Comment:
Please create a single context.Context and use that throughout instead of context.Background in every spot:
```go
ctx := context.Background()
stats.RecordWithTags(ctx, ...)
```

Reply:
Could you also ensure that you add a blank value intentionally, so that we can have `HasValue: true` with a `Value: ""`?

Reply:
The following are the examples of blank values for tags.
```
tag.Upsert(keyField, ""), // Empty field name
tag.Upsert(keyName, ""),  // Empty sprint name
```
Opencesus does not include tags whose values are blank. So there is never going to be a situation where `Value: ""` and `HasValue: true`

Reply:
I believe that's a mistake to not include `Value: ""` and `HasValue: true` because Stackdriver I believe supports it and so do other backends and we even have such values in specs if I recall correctly. Ignoring such values is going to cause a deviation when converting metrics from other sources (aka [receivers](https://opencensus.io/service/receivers/)) directly into OpenCensus Agent metrics.

Reply:
Actually, never mind. The various backends will deal with missing values themselves. We can come back and report her if we notice a deviation.